Tests/Testably.Abstractions.Tests/FileSystem/FileSystemWatcher/NotifyFiltersTests.cs

```diff
@@ -682,18 +684,61 @@ public async Task NotifyFilter_MoveDirectoryOutOfTheWatchedDirectory_ShouldTrigg
 		};
 
 		fileSystemWatcher.NotifyFilter = notifyFilter;
-		fileSystemWatcher.IncludeSubdirectories = true;
+		fileSystemWatcher.IncludeSubdirectories = includeSubdirectories;
 		fileSystemWatcher.EnableRaisingEvents = true;
 
 		FileSystem.Directory.Move(sourcePath, destinationName);
 
 		await That(ms.Wait(ExpectSuccess, TestContext.Current.CancellationToken)).IsTrue();
 		await That(result).IsNotNull();
 		await That(result!.ChangeType).IsEqualTo(WatcherChangeTypes.Deleted);
-		await That(result.FullPath).IsEqualTo(sourcePath);
+		await That(result.FullPath).IsEqualTo(FileSystem.Path.GetFullPath(sourcePath));
 		await That(result.Name).IsEqualTo(sourceName);
 	}
 
+	[Theory]
+	[InlineAutoData(NotifyFilters.DirectoryName, true)]
+	[InlineAutoData(NotifyFilters.DirectoryName, false)]
+	public async Task NotifyFilter_MoveDirectoryInToTheWatchedDirectory_ShouldTriggerCreatedEventOnNotifyFilters_OnWindows(
```

Comment:
In [your comment](https://github.com/Testably/Testably.Abstractions/issues/865#issuecomment-3299468675) you also mentioned files. Maybe we should also add test cases when files are moved in/out of the watched directory?